scipy/special/boost_special_functions.h

```diff
@@ -951,26 +951,46 @@ ncf_kurtosis_excess_double(double v1, double v2, double l)
 
 template<typename Real>
 Real
-nct_cdf_wrap(const Real x, const Real v, const Real l)
+nct_cdf_wrap(const Real v, const Real l, const Real x)
 {
-    if (std::isfinite(x)) {
-        return boost::math::cdf(
-            boost::math::non_central_t_distribution<Real, StatsPolicy>(v, l), x);
+    if (std::isnan(x) || std::isnan(v) || std::isnan(l)) {
+	return NAN;
     }
-    // -inf => 0, inf => 1
-    return 1.0 - std::signbit(x);
+    if (v <= 0) {
+	sf_error("nctdtr", SF_ERROR_DOMAIN, NULL);
+	return NAN;
+    }
+    if (std::isinf(x)) {
+	return 1.0;
+    }
+    Real y;
+    try {
+	y = boost::math::cdf(
+                boost::math::non_central_t_distribution<Real, SpecialPolicy>(v, l), x);
```

Comment:
I guess this is a more general question than being specific to this PR, but would it be possible to tailor `SpecialPolicy` to align with `scipy.special`’s conventions and  eliminate the need to `catch (...)` at each invocation?

Reply:
We catch here and call `sf_error` with the correct error code so that error handling can be controlled by [special.errstate](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.errstate.html) and [special.seterr](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.seterr.html#scipy.special.seterr).

Reply:
Indeed I see `ncf_ppf_wrap` of https://github.com/scipy/scipy/pull/21505/files properly “translates” `boost` exceptions to `special` error codes.